app/src/main/java/org/fossasia/openevent/app/module/event/dashboard/EventDashboardPresenter.java

```diff
@@ -74,6 +79,22 @@ public void loadDetails(boolean forceReload) {
             }, Logger::logError);
     }
 
+    @Override
+    public void toggleState() {
+        event.state = Event.STATE_DRAFT.equals(event.state) ? Event.STATE_PUBLISHED : Event.STATE_DRAFT;
+        eventRepository.updateEvent(event)
+            .compose(dispose(getDisposable()))
+            .compose(erroneous(getView()))
+            .doFinally(() -> getView().showResult(event))
```

Comment:
Show spinner and use ViewTransformers.showResult, because this is not backed by the job. The toggling should be synchronous with server and the toggle should only changed once it is synced up with it